sconfig/shared/src/main/scala/org/ekrich/config/impl/SimpleConfigObject.scala

```diff
@@ -501,81 +500,129 @@ final class SimpleConfigObject(
         v.relativized(prefix)
     })
 
-  override def render(
+  private def trySimplifyTheOnlyNestedObjectRec(
+      keysAggregate: String
+  ): Option[(String, AbstractConfigValue)] =
+    if (value.size() == 1) {
+      val newKeyElement = ConfigImplUtil.renderStringUnquotedIfPossible(
+        keySet.iterator().next()
+      )
+      val newAggregate = (if (keysAggregate.isEmpty) ""
+                          else s"$keysAggregate.") + newKeyElement
+      values.iterator().next() match {
+        case innerSCO: SimpleConfigObject =>
+          innerSCO.trySimplifyTheOnlyNestedObjectRec(newAggregate)
+        case other: AbstractConfigValue =>
+          Some(newAggregate -> other)
+        case _ => None
+      }
+    } else None
+
+  private def trySimplifyTheOnlyNestedObject(
+      options: ConfigRenderOptions
+  ): Option[(String, AbstractConfigValue)] =
+    if (!options.formattingOptions.simplifyOneEntryNestedObjects || options.getJson) {
+      None
+    } else trySimplifyTheOnlyNestedObjectRec("")
+
+  override def renderValue(
       sb: jl.StringBuilder,
       indentVal: Int,
       atRoot: Boolean,
       options: ConfigRenderOptions
   ): Unit = {
     if (isEmpty) sb.append("{}")
     else {
-      val outerBraces = options.getJson || !atRoot
-      var innerIndent = 0
-      if (outerBraces) {
-        innerIndent = indentVal + 1
-        sb.append("{")
-        if (options.getFormatted) sb.append('\n')
-      } else innerIndent = indentVal
-      var separatorCount = 0
-
-      val keys = new ju.ArrayList[String]
-      keys.addAll(keySet)
-      val ordering =
-        if (options.formattingOptions.keepOriginOrder)
-          new SimpleConfigObject.KeepOriginRenderComparator(str =>
-            value.get(str).origin
-          )
-        else new SimpleConfigObject.OrderedRenderComparator
-      ju.Collections.sort(keys, ordering)
-
-      for (k <- keys.asScala) {
-        var v: AbstractConfigValue = null
-        v = value.get(k)
-        if (options.getOriginComments) {
-          val lines = v.origin.description.split("\n")
-          for (l <- lines) {
-            AbstractConfigValue.indent(sb, indentVal + 1, options)
-            sb.append('#')
-            if (!l.isEmpty) sb.append(' ')
-            sb.append(l)
-            sb.append("\n")
+      trySimplifyTheOnlyNestedObject(options) match {
+        case Some((aggKey, leafValue)) =>
+          if (!atRoot) { // if not root then it is in some other object
+            if (options.getFormatted)
+              sb.deleteCharAt(
```

Comment:
nasty, better solution welcomed

Reply:
It looks pretty complex and I would have to spend lots of time to review to come up with something better. Are you ok with it as is for now? We could merge and then put in a ticket to improve.

Reply:
yes :)